presto-orc/src/test/java/com/facebook/presto/orc/TestSelectiveOrcReader.java

```diff
@@ -555,6 +568,166 @@ public void testDecimalSequence()
                                 false, true, 28999999999L, 28999999999L, false, true, true))));
     }
 
+    @Test
+    public void testSliceVarchar()
+            throws Exception
+    {
+        //dictionary
+        tester.testRoundTrip(VARCHAR, newArrayList(limit(cycle(ImmutableList.of("apple", "apple pie", "apple\uD835\uDC03", "apple\uFFFD")), 3000)));
```

Comment:
@bhhari testRoundTrip takes multiple filters; first it runs with no filter; then it runs once again using each provided filter; hence, you can combine multiple calls into one like so:

```
        //dictionary
        tester.testRoundTrip(VARCHAR, 
                newArrayList(limit(cycle(ImmutableList.of("apple", "apple pie", "apple\uD835\uDC03", "apple\uFFFD")), 3000)),
                BytesRange.of("apple".getBytes(), false, "apple".getBytes(), false, false));
        
        //direct
        tester.testRoundTrip(VARCHAR,
                createList(NUM_ROWS, String::valueOf),
                BytesValues.of(new byte[][] {"10".getBytes(), "11".getBytes()}, false),
                BytesValues.of(new byte[][] {"10".getBytes(), "11".getBytes()}, true),
                BytesRange.of("10".getBytes(), false, "14".getBytes(), false, false));
```